src/main/java/com/fabriccommunity/spookytime/world/feature/LargeDeadwoodTreeFeature.java

```diff
@@ -0,0 +1,198 @@
+package com.fabriccommunity.spookytime.world.feature;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.MutableIntBoundingBox;
+import net.minecraft.world.ModifiableTestableWorld;
+import net.minecraft.world.TestableWorld;
+import net.minecraft.world.gen.feature.AbstractTreeFeature;
+import net.minecraft.world.gen.feature.DefaultFeatureConfig;
+import com.mojang.datafixers.Dynamic;
+
+import com.fabriccommunity.spookytime.registry.SpookyBlocks;
+
+import java.util.Random;
+import java.util.Set;
+import java.util.function.Function;
+
+public class LargeDeadwoodTreeFeature extends AbstractTreeFeature<DefaultFeatureConfig> {
+	private static final BlockState LOG = SpookyBlocks.DEADWOOD_LOG.getDefaultState();
+	private static final BlockState LEAVES = SpookyBlocks.DEADWOOD_LEAVES.getDefaultState();
+	
+	public LargeDeadwoodTreeFeature(Function<Dynamic<?>, ? extends DefaultFeatureConfig> function, boolean emitBlockUpdates) {
+		super(function, emitBlockUpdates);
+	}
+	
+	protected static boolean isNaturalDirt(TestableWorld world, BlockPos pos) {
+		return world.testBlockState(pos, (state) -> {
+			Block block = state.getBlock();
+			return block == SpookyBlocks.DECEASED_DIRT;
+		});
+	}
+	
+	protected static boolean isNaturalDirtOrGrass(TestableWorld world, BlockPos pos) {
+		return world.testBlockState(pos, (state) -> {
+			Block block = state.getBlock();
+			return block == SpookyBlocks.DECEASED_DIRT || block == SpookyBlocks.DECEASED_GRASS_BLOCK;
+		});
+	}
+	
+	public boolean generate(Set<BlockPos> posSet, ModifiableTestableWorld world, Random random, BlockPos pos, MutableIntBoundingBox bb) {
+		int height = random.nextInt(3) + random.nextInt(2) + 6;
+		int x = pos.getX();
+		int y = pos.getY();
+		int z = pos.getZ();
+		if (y >= 1 && y + height + 1 < 256) {
+			BlockPos downPos = pos.down();
+			if (!isNaturalDirtOrGrass(world, downPos)) {
+				return false;
+			} else if (!this.doesTreeFit(world, pos, height)) {
+				return false;
+			} else {
+				this.setToDirt(world, downPos);
+				this.setToDirt(world, downPos.east());
+				this.setToDirt(world, downPos.south());
+				this.setToDirt(world, downPos.south().east());
+				Direction dir = Direction.Type.HORIZONTAL.random(random);
+				int g = height - random.nextInt(4);
+				int h = 2 - random.nextInt(3);
```

Comment:
I assume g and h are to do with leaf placement or something? Can we have more descriptive names?